plutus-tutorial/tutorial/Tutorial/02-validator-scripts.md

```diff
@@ -0,0 +1,205 @@
+    -- discards its result. If you want to hold on to the transaction you can
+    -- use 'W.payToScript'.
+    W.payToScript_ W.defaultSlotRange gameAddress (Ada.toValue adaValue) (mkDataScript word)
+```
+
+The second endpoint, `guess`, creates a transaction that spends the game output using the guessed word as a redeemer. 
+
+```haskell
+-- | The "guess" contract endpoint. See note [Contract endpoints]
+guess :: MonadWallet m => String -> m ()
+guess word =
+    -- 'collectFromScript' is a function of the wallet API. It consumes the
+    -- unspent transaction outputs at a script address and pays them to a
+    -- public key address owned by this wallet. It takes the validator script
+    -- and the redeemer scripts as arguments.
+    W.collectFromScript W.defaultSlotRange gameValidator (mkRedeemerScript word)
+```
+
+If we run `guess` now, nothing will happen. Why? Because in order to spend all outputs at the script address, the wallet needs to be aware of this address before the outputs are produced. That way, it can scan incoming blocks from the blockchain for outputs at that address, and doesn't have to keep a record of all unspent outputs of the entire blockchain. So before the game starts, players need to run the following action:
+
+```haskell
+-- | The "startGame" contract endpoint, telling the wallet to start watching
+--   the address of the game script.
+startGame :: MonadWallet m => m ()
+startGame =
+    -- 'startWatching' is a function of the wallet API. It instructs the wallet
+    -- to keep track of all outputs at the address. Player 2 needs to call
+    -- 'startGame' before Player 1 uses the 'lock' endpoint, to ensure that
+    -- Player 2's wallet is aware of the game address.
+    W.startWatching gameAddress
+```
+
+Endpoints can have any number of parameters: `lock` has two  parameters, `guess` has one and `startGame` has none. For each endpoint we include a call to `mkFunction` at the end of the contract definition, by writing `$(mkFunction 'lock)`, `$(mkFunction 'guess)` and `$(mkFunction 'startGame)` in three separate lines. This causes the Haskell compiler to generate a schema for the endpoint. The Plutus Playground then uses this schema to present an HTML form to the user where the parameters can be entered.
+
+# 2. Testing the contract in the Playground
+
+To test this contract, open the [Plutus Playground](https://prod.playground.plutus.iohkdev.io/) and click the "Game" button above the editor field. Then click "Compile". 
+
+You can now create a trace using the endpoints `lock`, `guess` and `startGame`. For a successful run of the game, click Wallet 1's `startGame` button, then Wallet 2's `lock` button and finally Wallet 1's `guess` button. Three boxes appear in the "Actions" section, numbered 1 to 3. In the second box, type "plutus" in the first input and 8 in the second input. In the third box type "plutus". The trace should look like the screenshot below.
+
+![A trace for the guessing game](game-actions.png)
```

Comment:
I have mixed feelings about checking in screenshots - they're hard to keep up to date...

Reply:
They are, but the text on its own would go out of date just as quickly